File: DSA Practice/LeetCode/202. Happy Number/solution.java
class Solution {
    public boolean isHappy(int n) {
        if (n == 1 || n == 7){
            return true;
        }
        
        while(n > 1){
            int temp = n;
            int ne = 0;
            while(temp > 0){
                int rem = temp % 10;
                ne += rem * rem;
                temp /= 10;
            }

            n = ne;

            if (n == 1 || n == 7){
                return true;
            } else if (n < 10){
                return false;
            }

        }

        return false;
    }
}

File: DSA Practice/LeetCode/202. Happy Number/solution.kt
class Solution {
    fun isHappy(n: Int): Boolean {
        var holder : Int = n
        if (holder == 1 || holder == 7){
            return true
        }

        while(holder > 1){
            var temp : Int = holder
            var s : Int = 0
            while(temp > 0){
                var rem : Int = temp%10
                s += rem*rem
                temp /= 10
            }

            holder = s

            if (holder == 1 || holder == 7){
                return true
            } else if (holder < 10){
                return false
            }
        }

        return false
    }
}